src/odemis/gui/cont/acquisition.py

```diff
@@ -2272,50 +2274,120 @@ def _on_ac_done(self, future):
 
 
 class FastEMAlignmentController:
-    def __init__(self, tab_data, panel):
+    """
+    Takes care of the calibration button and process in the calibration panel in the FastEM acquisition tab.
+    """
+    def __init__(self, tab_data, tab_panel):
+        """
+        tab_data: (FastEMAcquisitionGUIData) The representation of the microscope GUI.
+        tab_panel: (wx.Frame) The calibration panel, which contains the calibration button,
+                    the gauge and the label of the gauge.
+        """
         self._tab_data = tab_data
-        self._panel = panel
+        self._main_data_model = tab_data.main
+        self._panel = tab_panel
+
+        self.btn_align = tab_panel.btn_align
+        self.gauge_progress = tab_panel.align_gauge_progress  # progress bar
+        self.gauge_label = tab_panel.align_lbl_gauge  # progress bar label
+        self.is_aligned = tab_data.main.is_aligned
 
-        self._panel.align_gauge_progress.Hide()
-        panel.btn_align.Bind(wx.EVT_BUTTON, self._on_btn_align)
-        tab_data.main.is_aligned.subscribe(self._on_align_state, init=True)
+        self.gauge_progress.Hide()  # hide progress bar
+        self._panel.Parent.Layout()
+        self.btn_align.Bind(wx.EVT_BUTTON, self.on_align)
+
+        # Check if we deal with a real or simulated microscope. If it is a simulator,
+        # we cannot run all calibrations yet.
+        if model.getComponent(role="mbsem") == "FASTEM-sim":  # it's a simulator
```

Comment:
I doubt that you can compare a component with a string, and that it'd be ever return `True`. Probably you forgot `.name`. 
Also, there is an easier way to get the microscope component: `model.getMicroscope()`.

BTW, it feels really easy to break. We normally never user the name as a way to change the behaviour. On the other hand, I cannot think of a better way to detect that it's a simulator. Maybe write `.name.lower().endswith("sim")`?
Could you add something like a `# HACK warning: we don't have an official way to detect a component is simulated, but here, we really need to know that it's simulated otherwise we can never simulate an acquisition`.

Reply:
yes indeed I forgot the `.name`. Had only tested it in the terminal. I did not find anything better as well. That is why I went for this one. I will add the warning! Hopefully we have a simulation for all calibrations some time and then we can just get rid of this one here.